### PR TITLE
configuration changes

### PR DESCRIPTION
**Pull request by Jafnee**

### CODE REVIEW

File: dump.rdb [hunk 1]
```diff
@@ -0,0 +1 @@
+REDIS0006�ܳC�Z��V
```
Comment (matiboy) [inline dump.rdb]: Please remove this file and/or add it to `.gitignore`

File: minion/understanding/api.py [hunk 2]
```diff
@@ -35,21 +36,26 @@ class XMLParser(object):
...
+    def get_url(self):
+        return ''
+
+    @minion.core.utils.functions.configuration_getter
+    def get_method(self):
+        return 'get'
+
     def _validate_configuration(self):
-        if not self.get_configuration('url'):
+        if not self.get_url():
```
Comment (matiboy) [inline minion/understanding/api.py]: Can you look for a method called `RequiresNonEmptyConfiguration` or similar, it should basically do that

Comment (Jafnee) [inline minion/understanding/api.py]: Alright

File: minion/understanding/api.py [hunk 3]
```diff
@@ -35,21 +36,25 @@ class XMLParser(object):
...
+        return ''
+
+    @minion.core.utils.functions.configuration_getter
+    def get_method(self):
+        return 'get'
+
     def _validate_configuration(self):
-        if not self.get_configuration('url'):
-            raise minion.core.components.exceptions.ImproperlyConfigured('URL is required for API command')
+        self.requires_non_empty_configuration('url')
```
Comment (Jafnee) [inline minion/understanding/api.py:50]: @matiboy now using `requires_non_empty_configuration`

**Comment by matiboy** [comment]
Great, thanks!

**Pull request merged**